packages/salesforcedx-vscode-core/src/util/orgUtil.ts

```diff
@@ -40,58 +41,61 @@ export async function checkForExpiredOrgs(orgList: OrgList) {
       daysUntilExpiration.getDate() + daysBeforeExpire
     );
 
-    const authInfoObjects = await orgList.getAuthInfoObjects();
-    if (!authInfoObjects) {
+    const orgAuthorizations = await AuthInfo.listAllAuthorizations();
+    if (!orgAuthorizations) {
       return;
     }
 
-    const orgsAboutToExpire = authInfoObjects!.filter((authInfoObject: FileInfo) => {
+    let results: string[] = [];
+    for (let i=0; i<orgAuthorizations.length; i++) {
+      const orgAuthorization = orgAuthorizations[i];
+
       // Filter out the dev hubs.
-      if (authInfoObject.isDevHub) {
-        return false;
+      if (orgAuthorization.isDevHub) {
+        continue;
       }
 
-      // Some dev hubs have isDevHub=false, but no expiration date, so filter them out.
-      if (!authInfoObject.expirationDate) {
-        return false;
+      const authFields = await getAuthFieldsFor(orgAuthorization.username);
+
+      // Some dev hubs have isDevHub=false but no expiration date, so filter them out.
+      if (!authFields.expirationDate) {
+        continue;
       }
 
       // Filter out the expired orgs.
-      const expirationDate = new Date(authInfoObject.expirationDate);
+      const expirationDate = new Date(authFields.expirationDate);
       if (expirationDate < today) {
-        return false;
+        continue;
       }
 
       // Now filter and only return the results that are within the 5 day window.
-      return expirationDate <= daysUntilExpiration;
-    });
+      if (expirationDate <= daysUntilExpiration) {
+        const aliasName = orgAuthorization.aliases && orgAuthorization.aliases.length > 0
+          ? orgAuthorization.aliases[0]
+          : orgAuthorization.username;
+
+        results.push(
+          nls.localize(
+            'pending_org_expiration_expires_on_message',
+            aliasName,
+            authFields.expirationDate
+          )
+        );
+      }
+    }
 
-    if (!orgsAboutToExpire || orgsAboutToExpire.length < 1) {
+    if (results.length < 1) {
       return;
     }
```

Comment:
this is much easier to grok.  Nice update.

Reply:
Agreed!